tools/GenerateCommitHashHeader.ps1

```diff
@@ -1,5 +1,35 @@
 # This script gets the current git commit hash and places it in a header file stamped to a wstring_view
+# If literally anything goes wrong in this script with regards to git, like not being in a git repo,
+# not being able to find the current branch on remote, not being able to figure out what branch you're on,
+# we'll just return the latest commit hash of remote master.
 
 Write-Output "constexpr std::wstring_view CurrentCommitHash{ L`"" | Out-File -FilePath "Generated Files\CurrentCommitHash.h" -Encoding ASCII -NoNewline
-git rev-parse HEAD  | Out-File -FilePath "Generated Files\CurrentCommitHash.h" -Encoding ASCII -Append -NoNewline
+
+$latestMasterHash = (git ls-remote https://github.com/microsoft/terminal.git --heads master).Split()[0]

```

Comment:
Hmmm..
thinking about this a bit more, I'd be more comfortable with a rev-parse or the hardcoded string "master" (if git fails to execute) rather than us hitting a server during build. I'd be worried about trying to hit github for all local builds :smile:

What do you think?

Reply:
like, rev-parse is really good for locking down the exact commit -- that's perfect :smile: just in case it fails, we need a sane fallback value

Reply:
That's fair, it prob shouldn't hit github no matter what, I originally wanted to just leave it as "master" but I figured getting the actual hash would be more in line with what the script says it does.

Reply:
My other concern was that your local branch might be ahead in commits than remote, so if it just did a `rev-parse HEAD`, the link wouldn't work.

Reply:
I'm not too worried about that one. Local developers hitting F5 can have a slightly torn result -- as long as the build agents always show the commit they were running against

Reply:
...! i wonder if there's a buildagent environment variable we can use for the SHA1 instead of having to ask git at all
then we only rev-parse on the local machine...

Reply:
omg that's genius, it turns out there actually is one!